de.fu_berlin.inf.dpp.server/src/de/fu_berlin/inf/dpp/server/session/ServerSessionContextFactory.java

```diff
@@ -19,5 +19,6 @@ public final void createNonCoreComponents(ISarosSession session,
         container.addComponent(FileActivityExecutor.class);
         container.addComponent(FolderActivityExecutor.class);
         container.addComponent(TextEditActivityExecutor.class);
+        container.addComponent(NegotiationHandler.class);
```

Comment:
As you attach this class here I highly suggest to implement Startable or move this handler into application scope.